courses/C/intermediate-level/ Preprocessor Directives/intro.md

```diff
@@ -0,0 +1,163 @@
+---
+id: lesson-1
+title: "Preprocessor Directives"
+sidebar_label: Preprocessor Directives
+sidebar_position: 1
+description: "Learn Preprocessor Directives in C"
+tags: [courses,intermediate-level,File Inclusion,Introduction]
+---
+  
+
+Preprocessor directives in C are instructions that are processed by the preprocessor before the actual compilation of the code begins. They are used to control the compilation process, define constants, include files, and more.
+
+#### Introduction to the C Preprocessor
+
+The C preprocessor is a tool that performs textual substitutions on the source code before it is compiled. It handles directives that are introduced with a `#` symbol and processes them to prepare the code for compilation.
+
+
+#### Flowchart: Conditional Compilation
+:::note
+1. **Start**
+2. **Check if macro is defined (`#ifdef`)**
+   - If defined, include the code between `#ifdef` and `#endif`
+   - If not, skip the code between `#ifdef` and `#endif`
+3. **Check additional conditions using `#elif`, `#else` if applicable**
+4. **End**
+:::
+
+```
+ +--------------------------+
+ | Start                    |
+ +--------------------------+
+            |
+            v
+ +--------------------------+
+ | Check if macro is        |
+ | defined (`#ifdef`)       |
+ +--------------------------+
+            |
+          /   \
+       Yes     No
+        |       |
+        v       v
++------------------+  +--------------------------+
+| Include code      |  | Skip code                |
+| between #ifdef    |  | between #ifdef and       |
+| and #endif        |  | #endif                   |
++------------------+  +--------------------------+
+        |
+        v
++--------------------------+
+ | Check additional         |
+ | conditions with #elif,   |
+ | #else if applicable     |
+ +--------------------------+
+        |
+        v
++--------------------------+
+ | End                      |
+ +--------------------------+
+```
```

Comment:
Use a mermaid diagram for flow chart for example https://mermaid.js.org/syntax/flowchart.html#minimum-length-of-a-link